bin/si-smoketest/binary_exeuction_lib.ts

```diff
@@ -0,0 +1,18 @@
+// This lib can be used for binary args management, i.e.
```

Comment:
I feel this code could still love inside our main code, since it won't be used anywhere else and is very coupled to test startup, but admittedly it's a nitpick

Reply:
It could, but I think that as we develop out the other options for execution this is going to get a bit distracting from the intent of main.ts. 

In my opinion, main.ts should really be focused on driving the tests through the test_execution library, rather than handling all the cases that aren't even valid, like some of these:

i.e. 
- you specify a ramp test but don't give it a profile to ramp against
- you specify two types of test execution at the same time
- you specify an output folder that can't exist or be written to by the executing process
- etc etc

Handling all these cases will be hundreds of lines of .ts I'd imagine, so I thought it would be worth segregating off early to save us a refactor.

Happy to move it back in line in main if you wish, your call 🙂

Reply:
I'm hyper aware of codebase fragmentation because this usually turns into the user who will debug the code later having to traverse 3 different files to find out a pretty simple algorithm that is only used in one place on the other side - I'd rather have the hundreds of lines in hand when deciding how to break it apart, you know?

That being sad, our priority is to get this running and this works perfectly fine so I won't block it at all, let's get this train running!